module/src/hooks/useWillUnmountEffect.ts

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+/** A useEffect which only runs on the cleanup of the last effect */
+export function useWillUnMountEffect(callback: () => void) {
+  React.useEffect(() => callback, []);
+}
+
+/** A useLayoutEffect which only runs on the cleanup of the last effect */
+export function useWillUnMountLayoutEffect(callback: () => void) {
+  React.useLayoutEffect(() => callback, []);
+}
```

Comment:
For the same reason as earlier, I would be in favour of this for completeness:
```
/** A useEffect which only runs on the cleanup of the last effect */
export function useWillUnMountEffect(callback: ReturnType<React.EffectCallback>) {
  return React.useEffect(() => callback, []);
}

/** A useLayoutEffect which only runs on the cleanup of the last effect */
export function useWillUnMountLayoutEffect(callback: ReturnType<React.EffectCallback>) {
  return React.useLayoutEffect(() => callback, []);
}
```

Reply:
That's groovy